### PR TITLE
supporting lazy-fpsimd in guest on Arm64

### PR DESCRIPTION
**Pull request by lubinszARM**
Several jobs were finished in this patch:
1, provide functions to get/set fpcr/fpsr/vregs
2, support lazy-fpsimd-context-switch in el1

Signed-off-by: Bin Lu <bin.lu@arm.com>

### CODE REVIEW

**Comment by lubinszARM** [comment]
Related issue: https://github.com/google/gvisor/issues/1238

**Comment by lubinszARM** [comment]
@avagin 
Any comments?

File: pkg/sentry/platform/ring0/defs_arm64.go [hunk 1]
```diff
@@ -73,6 +73,9 @@ type CPUArchState struct {
 
 	// application context pointer.
 	appAddr uintptr
+
+	// lazyVfp is the value of cpacr_el1.
+	lazyVfp uintptr
```
Comment (prattmic) [inline pkg/sentry/platform/ring0/defs_arm64.go]: lazyVFP

https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: pkg/sentry/platform/ring0/lib_arm64.go [hunk 2]
```diff
@@ -16,10 +16,27 @@
...
 
-// SaveFloatingPoint saves floating point state by the most efficient mechanism
-// available (set by Init).
-var SaveFloatingPoint func(*byte)
+// FPCR, Floating-point Control Register.
+// Get the vaule of FPCR.
+func GetFpcr() (value uintptr)
+
+// Set the value of FPCR.
+func SetFpcr(value uintptr)
```
Comment (prattmic) [inline pkg/sentry/platform/ring0/lib_arm64.go]: These are all initialisms, so they should be SetFPCR, etc.

https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: pkg/sentry/platform/ring0/lib_arm64.go [hunk 3]
```diff
@@ -16,10 +16,27 @@
 
 package ring0
 
-// LoadFloatingPoint loads floating point state by the most efficient mechanism
-// available (set by Init).
-var LoadFloatingPoint func(*byte)
+// CPACR_EL1, Architectural Feature Access Control Register.
+// Get the value of CPACR_EL1.
+func GetCpacrEl1() (value uintptr)
```
Comment (prattmic) [inline pkg/sentry/platform/ring0/lib_arm64.go]: This should be named CPACREL1. Not need for 'Get'.

FWIW, with the proper capitalization, this is admittedly not super readable. It looks like there is no CPACR_EL2. Would it make sense to just name this CPACR?

https://golang.org/doc/effective_go.html#Getters

File: pkg/sentry/platform/ring0/lib_arm64.go [hunk 4]
```diff
@@ -16,10 +16,27 @@
 
 package ring0
 
-// LoadFloatingPoint loads floating point state by the most efficient mechanism
-// available (set by Init).
-var LoadFloatingPoint func(*byte)
+// CPACR_EL1, Architectural Feature Access Control Register.
```
Comment (prattmic) [inline pkg/sentry/platform/ring0/lib_arm64.go]: Doc comments should be a sentence starting with the name of the function. e.g., "CPACREL1 returns the value of the CPACR_EL1 register."

https://golang.org/doc/effective_go.html#commentary

File: pkg/sentry/platform/ring0/lib_arm64.go [hunk 5]
```diff
@@ -16,10 +16,27 @@
...
+// Get the value of CPACR_EL1.
+func GetCpacrEl1() (value uintptr)
 
-// SaveFloatingPoint saves floating point state by the most efficient mechanism
-// available (set by Init).
-var SaveFloatingPoint func(*byte)
+// FPCR, Floating-point Control Register.
+// Get the vaule of FPCR.
```
Comment (prattmic) [inline pkg/sentry/platform/ring0/lib_arm64.go]: value

File: pkg/sentry/platform/ring0/defs_arm64.go (hunk 1 above)
Comment (lubinsz) [reply pkg/sentry/platform/ring0/defs_arm64.go]: Done.

File: pkg/sentry/platform/ring0/lib_arm64.go (hunk 2 above)
Comment (lubinsz) [reply pkg/sentry/platform/ring0/lib_arm64.go]: Done.

File: pkg/sentry/platform/ring0/lib_arm64.go (hunk 3 above)
Comment (lubinsz) [reply pkg/sentry/platform/ring0/lib_arm64.go]: Done.
CPACR is a Arm32 register. On Aarch64, the name is cpacr_el1.
Please check it: https://developer.arm.com/docs/ddi0595/b/aarch32-system-registers/cpacr

File: pkg/sentry/platform/ring0/lib_arm64.go (hunk 4 above)
Comment (lubinsz) [reply pkg/sentry/platform/ring0/lib_arm64.go]: Done.

File: pkg/sentry/platform/ring0/lib_arm64.go (hunk 5 above)
Comment (lubinsz) [reply pkg/sentry/platform/ring0/lib_arm64.go]: Done.

**Pull request merged**